test/unittest/dds/subscriber/DataReaderTests.cpp

```diff
@@ -1295,6 +1301,148 @@ TEST_F(DataReaderTests, SetListener)
+    }
+
+    MockConsumer* mockConsumer;
+
+    const uint32_t AsyncTries = 5;
+    const uint32_t AsyncWaitMs = 25;
+
+    void HELPER_WaitForEntries(
+            uint32_t amount)
+    {
+        size_t entries = 0;
+        for (uint32_t i = 0; i != AsyncTries; i++)
+        {
+            entries = mockConsumer->ConsumedEntries().size();
+            if (entries == amount)
+            {
+                break;
+            }
+            std::this_thread::sleep_for(std::chrono::milliseconds(AsyncWaitMs));
+        }
+
+        ASSERT_EQ(amount, mockConsumer->ConsumedEntries().size());
+    }
+
+};
+
+/*
+ * This test checks that the DataReader methods defined in the standard not yet implemented in FastDDS return
+ * ReturnCode_t::RETCODE_UNSUPPORTED. The following methods are checked:
+ * 1. copy_from_topic_qos
+ * 2. delete_contained_entities
+ */
+TEST_F(DataReaderUnsuportedTests, UnsupportedDataReaderMethods)
+{
+    DomainParticipant* participant =
+            DomainParticipantFactory::get_instance()->create_participant(0, PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
+    Subscriber* subscriber = participant->create_subscriber(SUBSCRIBER_QOS_DEFAULT);
+    ASSERT_NE(subscriber, nullptr);
+
+    TypeSupport type(new FooTypeSupport());
+    type.register_type(participant);
+
+    Topic* topic = participant->create_topic("footopic", type.get_type_name(), TOPIC_QOS_DEFAULT);
+    ASSERT_NE(topic, nullptr);
+
+    DataReader* data_reader = subscriber->create_datareader(topic, DATAREADER_QOS_DEFAULT);
+    ASSERT_NE(subscriber, nullptr);
+
+    {
+        SampleLostStatus status;
+        EXPECT_EQ(ReturnCode_t::RETCODE_UNSUPPORTED, data_reader->get_sample_lost_status(status));
+    }
+
+    {
+        SampleRejectedStatus status;
+        EXPECT_EQ(ReturnCode_t::RETCODE_UNSUPPORTED, data_reader->get_sample_rejected_status(status));
+    }
+
+    {
+        SubscriptionMatchedStatus status;
+        EXPECT_EQ(ReturnCode_t::RETCODE_UNSUPPORTED, data_reader->get_subscription_matched_status(status));
+    }
+
+    {
+        SubscriptionMatchedStatus status;
+        EXPECT_EQ(ReturnCode_t::RETCODE_UNSUPPORTED, data_reader->get_subscription_matched_status(status));
+    }
+
+    // Tests require unimplemented struct PublicationBuiltinTopicData and classes ReadCondition and QueryCondition
+    //
+    // PublicationBuiltinTopicData publication_data;
+    // fastrtps::rtps::InstanceHandle_t publication_handle;
+    // EXPECT_EQ(
+    //        ReturnCode_t::RETCODE_UNSUPPORTED,
+    //        data_reader->get_matched_publication_data(publication_data,publication_handle));
+
+    // {
+    //     std::vector<SampleStateKind> sample_states;
+    //     std::vector<ViewStateKind> view_states;
+    //     std::vector<InstanceStateKind> instance_states;
+    //     EXPECT_EQ(
+    //         nullptr,
+    //         data_reader->create_readcondition(sample_states, view_states, instance_states));
```

Comment:
I think this method can be call as it is. No class implementation is needed to create a vector (empty one).